### ISSUE TITLE
:fire: Reconsider release date

### ISSUE THREAD

**Issue by alexandernst** [atom/atom#7308]
Big title with fire on it. Did I get your attention?

I want to beg you, dear Atom developers, to reconsider the release date (whenever that is) of 1.0.
I keep seeing issues/tasks that get their `1.0-roadmap` tag removed or just partially fixed because <i>there is no enough time before 1.0 to properly fix this™</i>.

Atom is still lacking some basic features that even Windows's Notepad has (an piece of software that, arguably, can be called a text editor).

Not supporting this or not implementing properly that in a hypothetical 1.0 release won't cause any good to Atom's reputation, probably won't cause any good to your feelings for your <i>baby</i> either when tens of posts start appearing, right after the 1.0 release, mocking about Atom not having X or Y.
(`Atom is the new Nano!`, `Atom, nice toy, not that nice editor`, etc... You get it)

I know 1.0 is not supposed to be <b>the</b> final and ultimate editor, but rather a point in time where the API is stable enough to be supported without BC until 2.0. I understand your point. But try to understand what <b>users</b> (not developers) would expect from a 1.0 release (which has been in the oven for quite some time now).

On the other hand, you left the biggest changes for the very end of the sprint towards 1.0 (that is, all the text-buffer and optimizations you have been doing in the past weeks). Not that I don't trust your skills or all the unit tests you have, but one would wait a prudential amount of time before releasing anything after such a big changes.

Finally, there are quite some issues that, imho, are very important (but post 1.0) and could potentially require big changes (like the char measurement system, related to ligatures in fonts). Not really sure how you're going to deal with those tasks if they require BC. Are you going to release a 2.0 just a few months later after releasing the 1.0? Or are you going to stop those BC (thus not fixing those tasks) for some time?

TL;DR: Don't rush with the release just because you agreed on releasing 1.0 on a specific date. It would be better to delay the release instead of releasing a half-baked thing.

**Comment by alexandernst** [comment]
Just some facts that I'd like to throw here:

* 53 issues related to how Atom is rendering text. https://github.com/atom/atom/labels/editor-rendering
* 21 issues related to crashes (from which only 7 need a repro): https://github.com/atom/atom/labels/crash <--- Keep in mind that "auto-restore buffers" is not implemented yet and from what @benogle said it won't happen before 1.0!
* 36 issues keep having the `1.0-roadmap` tag. https://github.com/atom/atom/labels/1.0-roadmap

**Comment by mark-hahn** [comment]
You said that you understand it is only an API freeze but then you go on to
show a bunch of bugs that have nothing to do with the API.  It is *just* an
API freeze.

**Comment by thedaniel** [comment]
We appreciate your concern, but the core team is OK with what's going to make it in this month as being 1.0, and we expect to continue the current pace of improvement to proceed in 1.0+. That is to say, we also think the stuff that you think is important is important and expect to hit it soon post-1.0.

Your point about negative feedback is valid. People have been writing (sometimes legit, sometimes spurious) sassy comments and articles about Atom since the day it came out, and user growth remains steadily up and to the right, so we have learned to accept that negative feedback exists while we focus on what we believe to be important.

**Comment by batjko** [comment]
Also, negative feedback is simply a fact of life for any new piece of software that competes with others, as comparisons are always drawn and personal preferences are held up as must-have requirements etc etc. 

And that's fine. It's an important type of feedback, and usually constructive albeit not necessarily very pleasant.

**Comment by mark-hahn** [comment]
Javascript/NodeJS has been criticized since day one and still is.  See
Hacker News.  The opinions/arguments are continually rehashed ad nauseam
with no new insights expressed in years.

I am continually amazed at how many users take one bug or missing feature
and get all worked up.  They say that the one problem keeps them from using
atom or that atom can't possibly continue without it.  They are really
missing the big picture.

I see Atom as an organic fast-evolving organism with warts but huge
opportunities.  Versions with major improvements seem to come out multiple
times a week.  It follows Google's modern fast release cycle methodology
compared to Microsoft's snail pace.

**Comment by alexandernst** [comment]
@mark-hahn Yes, I do understand that 1.0 is just an API freeze. What I can't understand is why tag the entire editor under the "1.0" mark instead of saying "Hey, the API is freezed now, come and make plugins for our editor". 

I'm worried that tagging Atom as 1.0 will harm it as the majority of users won't see the 1.0 release as "API is freezed", but rather as "we think you'll love our product". And, let's be honest, Atom is nowhere near perfect. Major improvements can be done in almost all aspects, from performance and other low-level stuff up to the UI/UX.

I'm amazed how you converted my issue about Atom into a JS defense and ... Google vs Microsoft release cycles.

Anyways, after discussing this with another Atom dev it's clear that a decision has been already made. Lets all hope for the best.

**Comment by mark-hahn** [comment]
> instead of saying "Hey, the API is freezed now, come and make plugins for
our editor".

I think most programmers nowadays understand the meaning of semver.

> how you converted my issue about Atom into a JS defense

Yes, I ramble.  The drift of the thread was towards a discussion of
negative criticism.  Maybe this topic should be on discuss.atom.io.


On Wed, Jun 17, 2015 at 10:34 AM, Daniel Hengeveld <notifications@github.com
